<launch>
  <!-- camera driver -->
  
  <node name="davis_ros_driver" pkg="davis_ros_driver" type="davis_ros_driver">
    <rosparam command="load" file="$(find davis_ros_driver)/config/indoors.yaml" />
  </node>

    <!-- feature detector -->
  <node name="visual_servo_node" pkg="slip_detection_davis" type="visual_servo_node" output="screen">



  </node> 

  <!-- corner visualization -->
  <node name="dvs_renderer_corner_soft" pkg="dvs_renderer" type="dvs_renderer">
    <param name="display_method" value="red-blue"/>
    <param name="num_event_msgs" value="5"/>
    <remap from="events" to="/dvs_corner_events_soft" />
    <remap from="image" to="/dvs/image_raw" />
    <remap from="camera_info" to="/dvs/camera_info" />
    <remap from="dvs_rendering" to="dvs_rendering_corners_soft"/>
  </node>

  <!-- event visualization -->
  <node name="dvs_renderer_edge_soft" pkg="dvs_renderer" type="dvs_renderer">
    <param name="display_method" value="red-blue"/>
    <param name="num_event_msgs" value="1"/>
    <remap from="events" to="/dvs_edge_events_soft" />
    <remap from="image" to="/dvs/image_raw" />
    <remap from="camera_info" to="/dvs/camera_info" />
    <remap from="dvs_rendering" to="dvs_rendering_edges_soft"/>
  </node>


  <!-- event visualization -->
  <node name="dvs_renderer_noise_soft" pkg="dvs_renderer" type="dvs_renderer">
    <param name="display_method" value="red-blue"/>
    <param name="num_event_msgs" value="1"/>
    <remap from="events" to="/dvs_noise_events_soft" />
    <remap from="image" to="/dvs/image_raw" />
    <remap from="camera_info" to="/dvs/camera_info" />
    <remap from="dvs_rendering" to="dvs_rendering_noises_soft"/>
  </node>


<node name="dvs_renderer_raw" pkg="dvs_renderer" type="dvs_renderer">
    <param name="display_method" value="red-blue"/>
    <param name="num_event_msgs" value="1"/>
    <remap from="events" to="/dvs/events" />
    <remap from="camera_info" to="/dvs/camera_info" />
    <remap from="dvs_rendering" to="dvs_renderer_raw"/>
  </node>

<node name="dvs_renderer_center" pkg="dvs_renderer" type="dvs_renderer">
    <param name="display_method" value="red-blue"/>
    <param name="num_event_msgs" value="1"/>
    <remap from="events" to="/object_center" />
    <remap from="camera_info" to="/dvs/camera_info" />
    <remap from="dvs_rendering" to="dvs_renderer_raw"/>
  </node>

  <!-- display -->
  <node name="rqt_gui_corners" pkg="rqt_gui" type="rqt_gui" args="--perspective-file $(find slip_detection_davis)/cfg/soft_exp.perspective" />



</launch>
